src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -15,290 +15,1566 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+            // using the natural type List<int>, even though ListBase<int>.Add(string) has the wrong parameter type.)
+            // Add similar test where ListBase<T> has no Add method. We should still use ListBase<int> for x and y.
+            var comp = CreateEmptyCompilation(new[] { sourceA, sourceB }, parseOptions: TestOptions.RegularPreview.WithNoRefSafetyRulesAttribute());
             comp.VerifyEmitDiagnostics(
-                // (5,9): error CS0411: The type arguments for method 'Program.F<T>(T)' cannot be inferred from the usage. Try specifying the type arguments explicitly.
-                //         F([]);
-                Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "F").WithArguments("Program.F<T>(T)").WithLocation(5, 9),
-                // (6,9): error CS0411: The type arguments for method 'Program.F<T>(T)' cannot be inferred from the usage. Try specifying the type arguments explicitly.
-                //         F([[]]);
-                Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "F").WithArguments("Program.F<T>(T)").WithLocation(6, 9));
+                // warning CS8021: No value for RuntimeMetadataVersion found. No assembly containing System.Object was found nor was a value for RuntimeMetadataVersion specified through options.
+                Diagnostic(ErrorCode.WRN_NoRuntimeMetadataVersion).WithLocation(1, 1),
+                // 1.cs(7,28): error CS1950: The best overloaded Add method 'ListBase<int>.Add(string)' for the collection initializer has some invalid arguments
+                //         ListBase<int> y = [1, 2];
+                Diagnostic(ErrorCode.ERR_BadArgTypesForCollectionAdd, "1").WithArguments("System.Collections.Generic.ListBase<int>.Add(string)").WithLocation(7, 28),
+                // 1.cs(7,28): error CS1503: Argument 1: cannot convert from 'int' to 'string'
+                //         ListBase<int> y = [1, 2];
+                Diagnostic(ErrorCode.ERR_BadArgType, "1").WithArguments("1", "int", "string").WithLocation(7, 28),
+                // 1.cs(7,31): error CS1950: The best overloaded Add method 'ListBase<int>.Add(string)' for the collection initializer has some invalid arguments
+                //         ListBase<int> y = [1, 2];
+                Diagnostic(ErrorCode.ERR_BadArgTypesForCollectionAdd, "2").WithArguments("System.Collections.Generic.ListBase<int>.Add(string)").WithLocation(7, 31),
+                // 1.cs(7,31): error CS1503: Argument 1: cannot convert from 'int' to 'string'
+                //         ListBase<int> y = [1, 2];
+                Diagnostic(ErrorCode.ERR_BadArgType, "2").WithArguments("1", "int", "string").WithLocation(7, 31));
         }
 
         [Fact]
-        public void NaturalType_05()
+        public void MissingList()
         {
-            string source = """
+            string sourceA = """
+                namespace System
```

Comment:
Can we not just use `MakeTypeMissing`/`MakeMemberMissing` (for the test below)?

Reply:
We could, and I have used `MakeTypeMissing()` and `MakeMemberMissing()` for other tests. But testing without the type at all seems useful.